extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/S3TaskLogs.java

```diff
@@ -67,57 +66,64 @@ public S3TaskLogs(
   public Optional<InputStream> streamTaskLog(final String taskid, final long offset) throws IOException
   {
     final String taskKey = getTaskLogKey(taskid, "log");
-    return streamTaskFile(offset, taskKey);
+    return streamTaskFileWithRetry(offset, taskKey);
   }
 
   @Override
   public Optional<InputStream> streamTaskReports(String taskid) throws IOException
   {
     final String taskKey = getTaskLogKey(taskid, "report.json");
-    return streamTaskFile(0, taskKey);
+    return streamTaskFileWithRetry(0, taskKey);
   }
 
   @Override
   public Optional<InputStream> streamTaskStatus(String taskid) throws IOException
   {
     final String taskKey = getTaskLogKey(taskid, "status.json");
-    return streamTaskFile(0, taskKey);
+    return streamTaskFileWithRetry(0, taskKey);
   }
 
-  private Optional<InputStream> streamTaskFile(final long offset, String taskKey) throws IOException
+  /**
+   * Using the retry conditions defined in {@link S3Utils#S3RETRY}.
+   */
+  private Optional<InputStream> streamTaskFileWithRetry(final long offset, String taskKey) throws IOException
+  {
+    try {
+      return S3Utils.retryS3Operation(() -> streamTaskFile(offset, taskKey));
+    }
+    catch (Exception e) {
+      throw new IOE(e, "Failed to stream logs from: %s", taskKey);
+    }
+  }
+
+  private Optional<InputStream> streamTaskFile(final long offset, String taskKey)
   {
     try {
       final ObjectMetadata objectMetadata = service.getObjectMetadata(config.getS3Bucket(), taskKey);
 
-      try {
-        final long start;
-        final long end = objectMetadata.getContentLength() - 1;
+      final long start;
+      final long end = objectMetadata.getContentLength() - 1;
 
-        if (offset > 0 && offset < objectMetadata.getContentLength()) {
-          start = offset;
-        } else if (offset < 0 && (-1 * offset) < objectMetadata.getContentLength()) {
-          start = objectMetadata.getContentLength() + offset;
-        } else {
-          start = 0;
-        }
+      long contentLength = objectMetadata.getContentLength();
+      if (offset >= contentLength || offset <= -contentLength) {
+        start = 0;
+      } else {
+        start = offset >= 0 ? offset : contentLength + offset;
+      }
```

Comment:
best not to mix ternary operator and if-else
```suggestion
      } else if (offset >= 0) {
        start = offset;
      } else {
        start = contentLength + offset;
      }
```